### ISSUE TITLE
Matching institution name with member code on Workshop Request Form

### ISSUE THREAD

**Issue by maneesha** [carpentries/amy#2581]
Now that we are validating member codes on the workshop request form, should we consider matching the institution name to the member code?  For example in this case, should the user get an error if `123456` is not the valid code for `Pizza Institute of Education` (even if it is a valid code otherwise)?  Should the admin get an error on their end when they try to accept the workshop request?  I am thinking about the user experience, the admin experience, as well as any data leaks this may cause.  

Note that without this, we could have instances where the institution name does not match the membership code. 

Tagging @danielle06 @elletjies  @sheraaronhurt  for feedback.

![image](https://github.com/carpentries/amy/assets/829690/e418b8d8-76f4-43b8-8cc7-65d909408789)

**Comment by elichad** [comment]
Sometimes members donate workshops to other institutions, so the member code may not always match the requester's institution. (Same for IT seats)

I talked a bit with the WAT about auto-filling the institution in the form based on the code, but the way that auto-fill works means all other options get hidden, which isn't ideal if the two shouldn't match and the user wants to override the choice. So I didn't implement this in the end.

I did add membership info to the workshop request details page on the admin side, which will hopefully help to avoid mismatches (see screenshot).
![Workshop request details page, showing the value of the member code field and an info box showing the name of the related membership and the number of workshops it has remaining.](https://github.com/carpentries/amy/assets/20683271/989bf7c7-2380-4754-bfe2-9ea7adf1b2cd)
